gapis/gfxapi/vulkan/templates/vk_spy_helpers.cpp.tmpl

```diff
@@ -421,6 +421,32 @@ void VulkanSpy::untrackMappedCoherentMemory(CallObserver*, uint64_t start, size_
 #endif // COHERENT_TRACKING_ENABLED
 }
 
+uint32_t VulkanSpy::SpyOverride_vkAllocateMemory(
+    VkDevice                     device,
+    VkMemoryAllocateInfo*  pAllocateInfo,
+    VkAllocationCallbacks* pAllocator,
+    VkDeviceMemory*              pMemory) {
+
+    uint32_t r = mImports.mVkDeviceFunctions[device].vkAllocateMemory(device, pAllocateInfo, pAllocator, pMemory);
+    std::shared_ptr<PhysicalDeviceObject> l_physical_device = PhysicalDevices[Devices[device]->mPhysicalDevice];
+    if (r >= 0 && 0 != (
+        l_physical_device->mMemoryProperties.mmemoryTypes[pAllocateInfo->mmemoryTypeIndex].mpropertyFlags &
+        ((uint32_t)(VkMemoryPropertyFlagBits::VK_MEMORY_PROPERTY_HOST_COHERENT_BIT)))) {
+        // This is host-coherent memory, we need to put in a hack to make some drivers behave with
+        // the memory tracker.
+        char* memory;
+        mImports.mVkDeviceFunctions[device].vkMapMemory(device, *pMemory, 0, pAllocateInfo->mallocationSize, 0, reinterpret_cast<void**>(&memory));
+        volatile char c;
+        for (size_t i = 0; i < pAllocateInfo->mallocationSize; ++i) {
+            c = *memory;
```

Comment:
Aak, I just noticed I was editing the .cpp file instead of the .tmpl file. That was the plan and i will fix.